NumberBox: add getting started description (#4377)
DevExpress/devextreme-documentation: concepts/05 UI Components/NumberBox/00 Getting Started with NumberBox/20 Handle Value Change.md

> Users can process a value after they enter it or after they click the spin buttons. Implement the [onValueChanged](/Documentation/ApiReference/UI_Components/dxNumberBox/Configuration/#onValueChanged) callback to handle value changes.

Reviewer: There are two more ways to change the NumberBox value.

```suggestion
Users can enter a new value or change the component value with spin buttons, Up and Down arrow keys, and mouse wheel. Implement the [onValueChanged](/Documentation/ApiReference/UI_Components/dxNumberBox/Configuration/#onValueChanged) callback to handle value changes.
```

> The following code displays a notification every time users change the NumberBox value.

Reviewer: ```suggestion
The following code displays a notification every time users change the NumberBox value:
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#number-box").dxNumberBox({
            // ...
            onValueChanged(e) {
                const currentNumber = e.value;
                if (currentNumber) {
                    DevExpress.ui.notify({
                        message: "The sum is $" + currentNumber
                    });
                }
            }
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-number-box ...

>         (onValueChanged) = "onValueChanged($event)"

Reviewer: ```suggestion
        (onValueChanged)="onValueChanged($event)"
```

    >
    </dx-number-box>

    <!-- tab: app.component.ts -->
    import { Component } from "@angular/core";
    import notify from "devextreme/ui/notify";

    @Component({
        selector: "app-root",
        templateUrl: "./app.component.html",
        styleUrls: ["./app.component.css"]
    })
    export class AppComponent {
        // ...

>         currentNumber: string;

Reviewer: Why string?

Author: By mistake

        onValueChanged(e) {

>             this.currentNumber = e.value;

Reviewer: You can two-way bind the NumberBox value to `currentNumber` and only use `e.value` to display notifications.

Author: Ok

            if (this.currentNumber) {
                notify({
                    message: "The sum is $" + this.currentNumber
                });
            }
        }
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxNumberBox ...

>             @value-changed = "onValueChanged($event)"

Reviewer: ```suggestion
            @value-changed="onValueChanged($event)"
```

        />
    </template>

    <script>
    // ...
    import { DxNumberBox } from "devextreme-vue/number-box";
    import notify from "devextreme/ui/notify";

    export default {
        components: {
            DxNumberBox
        },
        data() {
            return {

>                 currentNumber: ""

Reviewer: Why string?

Author: By mistake

            };
        },
        methods: {
            onValueChanged(e) {

>                 this.currentNumber = e.value;

Reviewer: You can two-way bind the NumberBox value to `currentNumber` and only use `e.value` to display notifications.

Author: Ok

                if (this.currentNumber) {
                    notify({
                        message: "The sum is $" + this.currentNumber
                    });
                }
            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React, { useCallback } from "react";
    import "devextreme/dist/css/dx.light.css";

    import { NumberBox } from "devextreme-react/number-box";
    import notify from "devextreme/ui/notify";

    function App() {
        const onValueChanged = useCallback((e) => {
            const currentNumber = e.value;
            if (currentNumber) {
                notify({
                    message: "The sum is $" + currentNumber
                });
            }
        }, []);

        return (
            <NumberBox ...

>                 onValueChanged = {onValueChanged}

Reviewer: ```suggestion
                onValueChanged={onValueChanged}
```
